canister/src/unstable_blocks.rs

```diff
@@ -170,32 +170,36 @@ pub fn get_chain_with_tip<'a, 'b>(
 
 // Returns the index of the `anchor`'s stable child if it exists.
 fn get_stable_child(blocks: &UnstableBlocks) -> Option<usize> {
-    // Compute the depth of all the children.
+    // Compute the difficulty based depth of all the children.
     let mut depths: Vec<_> = blocks
         .tree
         .children
         .iter()
         .enumerate()
-        .map(|(idx, child)| (blocktree::depth(child), idx))
+        .map(|(idx, child)| (blocktree::difficulty_based_depth(child), idx))
         .collect();
 
     // Sort by depth.
     depths.sort_by_key(|(depth, _child_idx)| *depth);
 
+    let root_difficulty = std::cmp::max(1, blocks.tree.root.difficulty()) as u128;
+
```

Comment:
Given that we are no longer dividing, do we need the `max(1, ..)` logic at this point?

Reply:
I mean we can remove max(), but in that case, if blocks.tree.root.difficulty() is zero, any block will be stable.

Reply:
I think that's fine, and expected, but maybe @THLO can confirm as well.

Reply:
As discussed with @ielashi offline, the [minimum difficulty](https://en.bitcoin.it/wiki/Difficulty) should be 1, so we shouldn't run into the situation where we need to deal with a difficulty of 0.